src/Avalonia.Controls/Flyouts/FlyoutBase.cs

```diff
@@ -0,0 +1,466 @@
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.BottomLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Bottom;
+                    break;
+
+                case FlyoutPlacementMode.LeftEdgeAlignedTop:
+                    _popup.PlacementRect = new Rect(0, 0, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.BottomLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Left;
+                    break;
+
+                case FlyoutPlacementMode.Left: //Left & centered
+                    _popup.PlacementRect = new Rect(0, 0, 1, trgtBnds.Height);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.Left;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.Left;
+                    break;
+
+                case FlyoutPlacementMode.LeftEdgeAlignedBottom:
+                    _popup.PlacementRect = new Rect(0, trgtBnds.Height - 1, 1, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.TopLeft;
+                    _popup.PlacementAnchor = PopupPositioning.PopupAnchor.BottomLeft;
+                    break;
+
+                case FlyoutPlacementMode.Full:
+                    //Not sure how the get this to work
+                    //Popup should display at max size in the middle of the VisualRoot/Window of the Target
+                    throw new NotSupportedException("FlyoutPlacementMode.Full is not supported at this time");
+                //break;
+
+                //includes Auto (not sure what determines that)...
+                default:
+                    //This is just FlyoutPlacementMode.Top behavior (above & centered)
+                    _popup.PlacementRect = new Rect(-sz.Width / 2, 0, sz.Width, 1);
+                    _popup.PlacementGravity = PopupPositioning.PopupGravity.Top;
+                    break;
+            }
+        }
+
+        private static void OnContextFlyoutPropertyChanged(AvaloniaPropertyChangedEventArgs args)
+        {
+            if (args.Sender is Control c)
+            {
+                if (args.OldValue is FlyoutBase)
+                {
+                    c.PointerReleased -= OnControlWithContextFlyoutPointerReleased;
+                }
+                if (args.NewValue is FlyoutBase)
+                {
+                    c.PointerReleased += OnControlWithContextFlyoutPointerReleased;
+                }
+            }
+        }
+
+        private static void OnControlWithContextFlyoutPointerReleased(object sender, PointerReleasedEventArgs e)
+        {
+            if (sender is Control c)
+            {
+                if (e.InitialPressMouseButton == MouseButton.Right &&
+                e.GetCurrentPoint(c).Properties.PointerUpdateKind == PointerUpdateKind.RightButtonReleased)
+                {
+                    if (c.ContextFlyout != null)
```

Comment:
Should we check if ContextMenu was set on the control, and ignore ContextFlyout if it was? To avoid breaking changes (let's say, user has set custom ContextFlyout on the text box, and with new update app will show two popups or only new one).
Maybe some warning in the user logs will be useful here as well.

Reply:
Add a check before showing ContextFlyout. If ContextMenu is also set, logs warning and defaults to ContextMenu.